blaze-core/src/test/scala/org/http4s/blazecore/websocket/WSTestHead.scala

```diff
@@ -74,28 +73,37 @@ sealed abstract class WSTestHead(
     * @param ws
     */
   def put(ws: WebSocketFrame): IO[Unit] =
-    inQueue.enqueue1(ws)
+    inQueue.offer(ws)
 
   val outStream: Stream[IO, WebSocketFrame] =
-    outQueue.dequeue
+    Stream.repeatEval(outQueue.take)
 
   /** poll our queue for a value,
     * timing out after `timeoutSeconds` seconds
     * runWorker(this);
     */
   def poll(timeoutSeconds: Long): IO[Option[WebSocketFrame]] =
-    IO.race(IO.sleep(timeoutSeconds.seconds), outQueue.dequeue1)
+    IO.race(IO.sleep(timeoutSeconds.seconds), outQueue.take)
       .map {
         case Left(_) => None
         case Right(wsFrame) =>
           Some(wsFrame)
       }
 
-  def pollBatch(batchSize: Int, timeoutSeconds: Long): IO[List[WebSocketFrame]] =
-    outQueue
-      .dequeueChunk1(batchSize)
-      .map(_.toList)
+  def pollBatch(batchSize: Int, timeoutSeconds: Long): IO[List[WebSocketFrame]] = {
+    def batch(acc: List[WebSocketFrame]): IO[List[WebSocketFrame]] =
```

Comment:
we should probably add a batched take operation to the CE queue, but it's getting mighty close to fs2 realm

Reply:
If Queue moved to CE3, then wouldn't the operation?

Reply:
CE3 implements a brand new `Queue`  separate from `fs2.concurrent.Queue`. CE3 doesn't have a concept of chunking (this was deemed an fs2 concern), but I think it probably makes sense to add something anyways

Originally the plan was for fs2 Queue to stick around but it seems like Mike made things work out with the CE3 one